src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -260,6 +287,111 @@ public String promptForPlatformVersion(List<String> versions) {
         return version;
 	}
 
+    @Override
+    public void promptForJdkPath(Server server) {
+
+        List<String> paths = new ArrayList<>();
+        paths.add("JAVA_HOME (currently: " + System.getProperty("java.home") + ")");
+        paths.addAll(getListOf5RecentJdkPaths());
+
+        String path = promptForMissingValueWithOptions("Which JDK would you like to use to run this server?",
+                server.getJdkPath(), "path", paths, true);
+
+        if (path.equals(paths.get(0))) {
+            server.setJdkPath(System.getProperty("java.home"));
+        }
+        else {
+            server.setJdkPath(path);
+        }
+    }
+
+    private void addJdkPathToSdkProperties(String path) {
+
+        String home = System.getProperty("user.home");
+        File openMRS = new File(home, SDKConstants.OPENMRS_SERVER_PATH + "/sdk.properties");
+
+        Writer output = null;
+        try {
+            output = new BufferedWriter(new FileWriter(openMRS, true));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        if (output != null) {
+            try {
+                if (!path.equals(System.getProperty("java.home")) && !isPathLineDuplicated(openMRS, path)) {
+                    output.write(path  + "\n");
+                    output.close();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        else {showMessage("Failed to save JDK path to properties file");
+        }
+    }
+
+    private boolean isPathLineDuplicated(File file, String jdkPath) {
+        LineIterator it = null;
+        try {
+            it = FileUtils.lineIterator(file, "UTF-8");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        if (it != null) {
+            while (it.hasNext()) {
+                if (jdkPath.equals(it.nextLine())) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean isThereJdkUnderPath(String jdkPath) {
+        File jdk = new File(jdkPath + "/bin/java");
+        if (jdk.exists()) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    @Override
+    public List<String> getListOf5RecentJdkPaths() {
+        final int count = 5;
+        String home = System.getProperty("user.home");
+        File openMRS = new File(home, SDKConstants.OPENMRS_SERVER_PATH + "/sdk.properties");
```

Comment:
See Server.getServersPathFile() and change to Properties sdkProperties = Server.getServersProperties();